Automattic/jetpack
# [RNMobile] Disable VideoPress settings for not belonged videos

## Proposed changes:

* Following the changes introduced in https://github.com/Automattic/jetpack/pull/30438 and https://github.com/Automattic/jetpack/pull/30443, disable Details, Privacy, and Rating settings if the video doesn't belong to the site.
* Render a warning message when the video doesn't belong to the site. I used the same design of the warning we display on the [`ContrastChecker` component](https://github.com/WordPress/gutenberg/blob/f0266c22cdbab064482c5e57d6a63a29b232cf2d/packages/block-editor/src/components/contrast-checker/index.native.js#L24-L59).

<details><summary><strong>Click here to show Android screenshots</strong></summary>

| Light mode | Dark mode |
|--------|--------|
|  <img src=https://github.com/Automattic/jetpack/assets/14905380/dccf8469-420f-4923-a74e-354a2de62cdc width=300> | <img src=https://github.com/Automattic/jetpack/assets/14905380/c47fb3bd-9001-4c5b-b0f2-abda758b63cd width=300> | 
| <img src=https://github.com/Automattic/jetpack/assets/14905380/73ee796d-68d6-4d53-9fed-f15501e29a89 width=300> | <img src=https://github.com/Automattic/jetpack/assets/14905380/41d8c4e3-99ae-498e-afa4-8161d72e8d1f width=300> | 

</details>

<details><summary><strong>Click here to show iOS screenshots</strong></summary>

| Light mode | Dark mode |
|--------|--------|
|  <img src=https://github.com/Automattic/jetpack/assets/14905380/87d32adf-199d-4fe7-be11-4c171e8dcfe0 width=300> | <img src=https://github.com/Automattic/jetpack/assets/14905380/7cd9681d-2cf9-41b5-8ac1-fa7f95a06775 width=300> | 
|  <img src=https://github.com/Automattic/jetpack/assets/14905380/3485f60e-0ddb-4004-a512-b7587b5e7e75 width=300> | <img src=https://github.com/Automattic/jetpack/assets/14905380/528c2369-167f-4126-bf84-dc7782502ca5 width=300> |
</details>

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


N/A

## Testing instructions:






* Open a post/page.
* Add a VideoPress block.
* Add a video.
* Open block settings and set title and description.
* Observe that it works as expected.
* Tap on the "Privacy and Rating" option.
* Change rating, privacy, "Allow download", and "Show video sharing menu" settings.
* Observe that it works as expected.
* Add another VideoPress block
* Add a video that doesn't belong to the site. E.g. insert the video from an URL - https://videopress.com/v/NGYE1Hqy.
* Open the block settings.
* Observe that the title and description settings are disabled and that a warning message is displayed.
* Tap on the "Privacy and Rating" option.
* Observe that settings are disabled and a warning message is displayed.
